js/lightbeam.js

```diff
@@ -1,9 +1,63 @@
+// transforms the object of nested objects 'websites' into a
+// usable format for d3
+function transformData(websites) {
+  const nodes = [];
+  const links = [];
+  const thirdParties = [];
```

Comment:
Okay I did most of this (the `addWebsite` function and used `set` instead of arrays). It is true that Google could be a first and third party; I've added logic to ensure a third party node isn't created if a first party node already exists (but the link is still made).

The reason for the `party` key is to flag first parties as circles and third parties as triangles when d3 goes to draw the elements onto the page. How else might this distinction be made?